File: src/main/java/ua/mezik/uchat/model/message/FileMessage.java
package ua.mezik.uchat.model.message;

import jakarta.persistence.Entity;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import ua.mezik.uchat.model.Account;
import ua.mezik.uchat.model.Chat;
import ua.mezik.uchat.model.ChatMessage;
import ua.mezik.uchat.model.MessageType;

@EqualsAndHashCode(callSuper = true)
@Entity
@Data
@NoArgsConstructor
public class FileMessage extends ChatMessage {
    private String link; // oh fuck you

    public FileMessage(Account owner, Chat chat, MessageType type, String link) {
        super(owner, chat, type);
        this.link = link;
    }
}

File: src/main/kotlin/com/mezik/uchat/model/database/FileMessage.kt
package com.mezik.uchat.model.database

import com.mezik.uchat.model.message.EncryptedMessage
import com.mezik.uchat.service.SequencedIdGenerator
import org.bson.types.ObjectId
import org.springframework.data.mongodb.core.mapping.Document

@Document("file_message")
class FileMessage(
    owner: Account,
    chat: Chat,
    type: MessageType,
    timestamp: Long,
    keyId: Long,
    val buffer: EncryptedMessage,
    databaseId: ObjectId = ObjectId(), // database requirement
    id: Long = SequencedIdGenerator.instance.generateId("chat_message")
) : ChatMessage(owner, chat, type, timestamp, keyId, databaseId = databaseId, id = id)
